src/test/resources/pacts/publicapi-direct-debit-connector-search-mandates-endpoint.json

```diff
@@ -0,0 +1,220 @@
+{
+  "consumer": {
+    "name": "publicapi"
+  },
+  "provider": {
+    "name": "connector"
+  },
+  "interactions": [
+    {
+      "description": "search mandates by parameters",
+      "providerStates": [
+        {
+          "name": "a gateway account with an external ids and a confirmed mandate with an email exists",
+          "params": {
+            "gateway_account_id": "2po9ycynwq8yxdgg2qwq9e9qpyrtre",
+            "mandate_id": "jkdjsvd8f78ffkwfek2a",
+            "unique_identifier": "AAAAA",
+            "charge_id": "charge97837509646393e3A",
+            "email": "rory@example.com"
+          }
+        }
+      ],
+      "request": {
+        "method": "GET",
+        "path": "/v1/api/accounts/123456/mandates",
+        "query": {
+          "email": [
+            "rory@example.com"
+          ]
+        }
+      },
+      "response": {
+        "status": 200,
+        "headers": {
+          "Content-Type": "application/json"
+        },
+        "body": {
+          "total": 1,
+          "count": 1,
+          "page": 1,
+          "results": [
+            {
+              "mandate_id": "obck2jlpu4erf3j2itfkp215ld",
+              "provider_id": "GOVERNMENTDI-ZY79J",
+              "reference": "aASasa",
+              "return_url": "https://www.example.com",
+              "payment_provider": "paymentProviderId",
+              "bank_statement_reference": "AAAA1111",
+              "created_date": "1995-10-27T10:37:19.753Z",
+              "payer": {
+                "name": "Mr Test",
+                "email": "rory@example.com"
+              },
+              "state": {
+                "status": "a status",
+                "details": "cause_details_from_gocardless"
+              },
+              "_links": {
+                "self": {
+                  "href": "https://publicapi.payments.service.gov.uk/v1/directdebit/mandates/obck2jlpu4erf3j2itfkp215ld",
+                  "method": "GET"
+                },
+                "events": {
+                  "href": "https://publicapi.payments.service.gov.uk/v1/directdebit/mandates/obck2jlpu4erf3j2itfkp215ld/events",
+                  "method": "GET"
+                },
+                "payments": {
+                  "href": "https://publicapi.payments.service.gov.uk/v1/directdebit/payments?mandate_id=obck2jlpu4erf3j2itfkp215ld",
+                  "method": "GET"
+                }
+              }
+            }
+          ],
+          "_links": {
+            "self": {
+              "href": "https://publicapi.payments.service.gov.uk/v1/directdebit/mandates?page=1&display_size=500"
```

Comment:
These links would include all of the query parameters that were provided, along with the page and display_size as you have here. The matcher doesn't care beyond `page=1.*` so maybe it isn't necessary to add them to this expected response but it could be miss leading for someone reading it? What do you think?